components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -318,4 +321,54 @@ void PageContentExtractor::GetSearchSummarizerKey(
   std::move(callback).Run(element.GetAttribute("content").Utf8());
 }
 
+void PageContentExtractor::ValidateOpenLeoButtonNonce(
+    const std::string& nonce,
+    mojom::PageContentExtractor::ValidateOpenLeoButtonNonceCallback callback) {
+  if (nonce.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto element = render_frame()->GetWebFrame()->GetDocument().GetElementById(
+      "continue-with-leo");
+  if (element.IsNull() || !element.HasHTMLTagName("a")) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  GURL url(element.GetAttribute("href").Utf8());
+  if (!IsOpenLeoButtonFromBraveSearchURL(url) ||
+      !element.HasAttribute("nonce") || url.ref_piece() != nonce) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  // Value of nonce property is not accessible via GetAttribute API, so we need
+  // to execute a script to get it.
+  blink::WebScriptSource source =
+      blink::WebScriptSource(blink::WebString::FromUTF8(
+          "document.getElementById('continue-with-leo').nonce"));
```

Comment:
Interesting, I wonder why that is - does it work with a `data-*` attribute (i.e. `data-nonce`)?

I think this would be more idiomatic anyway, as `data-*` attributes are namespaced for custom attributes

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/nonce#retrieving_a_nonce_value
This is what I found which I think is probably the reason to this.

Reply:
Okay, that's interesting - I guess fine as is though.